src/Workspaces/CoreTest/Differencing/LongestCommonSubsequenceTests.cs

```diff
@@ -0,0 +1,311 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Xunit;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Differencing.UnitTests
+{
+    public class LongestCommonSubsequenceTests
+    {
+        LongestCommonSubsequenceString lcs = new LongestCommonSubsequenceString();
+
+        private class LongestCommonSubsequenceString : LongestCommonSubsequence<string>
```

Comment:
Since the test is for the abstract LongestCommonSubsequence class it seems better to me to have a private class derived from it just for the testing rather then using the existing internal LongestCommonSubstring class. Just in case the internal class is changed in the future to not use the diff code from LongestCommonSubsequence.